lib/api/api_auth_webauthn.go

```diff
@@ -284,12 +286,12 @@ func (s *webauthnService) finishWebauthnRegistration(guiCfg config.GUIConfigurat
 		}
 		s.credentialsPendingRegistration = append(s.credentialsPendingRegistration, configCred)
 
-		credentialVolState := s.loadVolatileState()
-		credentialVolState.Credentials[configCred.ID] = WebauthnCredentialVolatileState{
-			SignCount:   credential.Authenticator.SignCount,
-			LastUseTime: now,
-		}
-		err = s.storeVolatileState(credentialVolState)
+		err = s.updateVolatileState(func(state *WebauthnVolatileState) {
```

Comment:
Why not use the new `updateCredentialVolatileState()` function here as well?  If I'm not mistaken, it will just as well add a previously non-existant `credId` with the same initial values?

Reply:
Good catch, thanks! As far as I can tell the only difference is that `CreateTime` and `LastUseTime` could get different values because this makes two `timeNow()` calls instead of using the same for both. Entirely insignificant.

This leaves only one use of `updateVolatileState`, so we _could_ merge that into `updateCredentialVolatileState`. But I think it still makes sense to keep the synchronization logic and the credential update logic as two separate components instead of mixing multiple responsibilities into the same function.